cmd/start/cmd.go

```diff
@@ -176,6 +166,27 @@ func enqueueNewBlocks(exportQueue types.HeightQueue, ctx *parser.Context) {
 	}
 }
 
+// mustGetLatestHeight will keep trying until it gets the latest height from RPC client, return 0 if retry count > 50
+func mustGetLatestHeight(ctx *parser.Context, retryCount int16) int64 {
+	if retryCount > 50 {
+		return 0
+	}
+
+	latestBlockHeight, err := ctx.Node.LatestHeight()
+	if err != nil {
+		avgBlockTime := config.Cfg.Parser.AvgBlockTime
+
+		ctx.Logger.Error("failed to get last block from RPCConfig client",
+			"err", err, "retry interval", avgBlockTime, "retry count", retryCount)
+		time.Sleep(avgBlockTime)
+
+		retryCount++
+		return mustGetLatestHeight(ctx, retryCount)
+	}
+
+	return latestBlockHeight
+}
```

Comment:
I think this can be simplified by making it iterative instead of recursive. This should allow you to not having to pass the initial `retryCount`:

```go
// mustGetLatestHeight tries getting the latest height from the RPC client.
// If after 50 tries no latest height can be found, it returns 0.
func mustGetLatestHeight(ctx *parser.Context) int64 {
	avgBlockTime := config.Cfg.Parser.AvgBlockTime
	for retryCount := 0; retryCount < 50; retryCount++ {
		latestBlockHeight, err := ctx.Node.LatestHeight()
		if err == nil {
			return latestBlockHeight
		}

		ctx.Logger.Error("failed to get last block from RPCConfig client",
			"err", err,
			"retry interval", avgBlockTime,
			"retry count", retryCount)
		time.Sleep(avgBlockTime)
	}

	return 0
}
```